mitmproxy/proxy/config.py

```diff
@@ -116,10 +116,10 @@ def __init__(
         self.openssl_method_server, self.openssl_options_server = \
             tcp.sslversion_choices[ssl_version_server]
 
-        if ssl_verify_upstream_cert:
-            self.openssl_verification_mode_server = SSL.VERIFY_PEER
-        else:
+        if ssl_insecure:
             self.openssl_verification_mode_server = SSL.VERIFY_NONE
+        else:
+            self.openssl_verification_mode_server = SSL.VERIFY_PEER
```

Comment:
Once we've added the attribute to the object here, we also need to make sure that the openssl_verification_mode_server attribute changes correctly when ssl_insecure changes. If we set this only in __init__, changing it in the console options will be a NOP.